core/execution/future/market.go

```diff
@@ -1940,6 +1940,10 @@ func (m *Market) submitOrder(ctx context.Context, order *types.Order) (*types.Or
 		return nil, nil, err
 	}
 
+	if err := m.position.ValidateOrder(ctx, order); err != nil {
```

Comment:
shouldn't we also validate in amend?
also, what happens if we have now (in the protocol upgrade snapshot) such an order already?

Reply:
True about amend, I'll have a look although that code path is much more complicated.

I've not checked mainnet snapshots yet to see if we have a bad one. I imagine the chances are slim because its only in the case where we reduce-only and the order `size` is massive, and so we'd have already seen a panic. In all other cases I think we would just fail at the margin check before we even got that far. Which is probably also true when amending.

Reply:
It should be quite unlikely, as these are the one used to close position and are reduce only. Because of this we don’t check margin for themS orders woth such a size which would have to go through margin check would surely not go through isn’t it?